### PR TITLE
add getWorkoutPlanId function

### PR DESCRIPTION
**Pull request by walterholohan**
WorkoutKit will be released in the next week with iOS17 which allows you to get a workout plan for an activity. I have created a separate function to get a workout plan id if you know the workout UUID. This is useful for match planned workouts to completed activities in Apple Health

### CODE REVIEW

File: ios/ReactNativeHealthkit.m [hunk 1]
```diff
@@ -211,7 +211,7 @@ @interface RCT_EXTERN_MODULE(ReactNativeHealthkit, RCTEventEmitter)
                   reject:(RCTPromiseRejectBlock)reject
 )
 
-RCT_EXTERN_METHOD(getWorkoutRoutes:(NSString)workoutUUID
+RCT_EXTERN_METHOD(getWorkoutPlanId:(NSString)workoutUUID
```
Comment (walterholohan) [inline ios/ReactNativeHealthkit.m:214]: rename this to the new function as it was a duplicate of the line above

**Comment by robertherber** [comment]
I'm guessing the reason it's not working currently is that it's [wrapped in a task](https://github.com/kingstinct/react-native-healthkit/blob/fc5f4174e07d04d06ad27268802b9cf139aa631d/ios/ReactNativeHealthkit.swift#L802) that resolves after the dict [has already been returned](https://github.com/kingstinct/react-native-healthkit/blob/fc5f4174e07d04d06ad27268802b9cf139aa631d/ios/ReactNativeHealthkit.swift#L819) to the JS side.

An extra method for just fetching an ID feels a bit verbose - or what do you think? Is there any data more than the ID itself that could be returned about the Workout Plan that could be interesting? But maybe this would fit better in it's very own query function (which could be either `getWorkoutPlanByWorkoutId` or `getWorkoutPlanById`)?

**Comment by walterholohan** [comment]
@robertherber thanks for the feedback I have now updated the function so it returns the workoutPlan (id and activityType fields for now)

**Comment by walterholohan** [comment]
@robertherber can I get another review please?

**Review by robertherber — APPROVED** [review]

**Pull request merged**